cpp/src/gandiva/regex_functions_holder.h

```diff
@@ -26,35 +26,38 @@
 #include "gandiva/execution_context.h"
 #include "gandiva/function_holder.h"
 #include "gandiva/node.h"
+#include "gandiva/regex_util.h"
 #include "gandiva/visibility.h"
 
 namespace gandiva {
 
-/// Function Holder for SQL 'like'
-class GANDIVA_EXPORT LikeHolder : public FunctionHolder {
+class GANDIVA_EXPORT RegexpMatchesHolder : public FunctionHolder {
```

Comment:
The name of the holder is a little bit confusing because it is responsible for other functions(like regexp_like).

Change it to `RegexpExpressionsHolder` or something similar